### PR TITLE
Fix #131, #132 and fix #133

### LINKED ISSUE #131: build: Build does not work with CMAKE_GENERATOR=Ninja in ENV
EDIT: see comment below, these are probably not correct steps to reproduce.

exporting `CMAKE_GENERATOR=Ninja` into shell which is used to build RoFI breaks the build

```bash
$ export CMAKE_GENERATOR=Ninja
$ . setup.sh -f Release
$ rcfg doc
$ rmake doc
Compiling suite doc
make: *** No targets specified and no makefile found.  Stop.
Build of suite doc failed. See output above
```

Either `rmake` should respect the generator used in CMake, or make sure `CMAKE_GENERATOR` is not set when `rcfg` invokes `cmake`.

### LINKED ISSUE #133: build: Documentation successfully configures but build fails if missing doxygen
It seems `rcfg doc` does not properly check if dependencies are available:

```bash
$ rcfg doc
Configuring suite doc
-- The C compiler identification is GNU 11.1.0
-- The CXX compiler identification is GNU 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: /home/xstill/repo/rofi/build.Release/doc

$ rmake doc
Compiling suite doc
[1/5] Extracting Doxygen for sim
FAILED: doc/doxygen/sim/xml/index.xml /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/sim/xml/index.xml 
cd /home/xstill/repo/rofi/build.Release/doc/doc && /home/xstill/repo/rofi/releng/doc/extractDoxygen.sh sim /home/xstill/repo/rofi/softwareComponents/rofiHalSim /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/sim
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 36: doxygen: command not found
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 37: doxygen: command not found
[2/5] Extracting Doxygen for lib
FAILED: doc/doxygen/lib/xml/index.xml /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/lib/xml/index.xml 
cd /home/xstill/repo/rofi/build.Release/doc/doc && /home/xstill/repo/rofi/releng/doc/extractDoxygen.sh lib /home/xstill/repo/rofi/softwareComponents /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/lib
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 36: doxygen: command not found
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 37: doxygen: command not found
[3/5] Extracting Doxygen for driver
FAILED: doc/doxygen/driver/xml/index.xml /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/driver/xml/index.xml 
cd /home/xstill/repo/rofi/build.Release/doc/doc && /home/xstill/repo/rofi/releng/doc/extractDoxygen.sh driver /home/xstill/repo/rofi/softwareComponents/rofiHalPhys /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/driver
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 36: doxygen: command not found
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 37: doxygen: command not found
[4/5] Extracting Doxygen for roficom
FAILED: doc/doxygen/roficom/xml/index.xml /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/roficom/xml/index.xml 
cd /home/xstill/repo/rofi/build.Release/doc/doc && /home/xstill/repo/rofi/releng/doc/extractDoxygen.sh roficom /home/xstill/repo/rofi/hardwareModules/RoFICoM/software/control_board /home/xstill/repo/rofi/build.Release/doc/doc/doxygen/roficom
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 36: doxygen: command not found
/home/xstill/repo/rofi/releng/doc/extractDoxygen.sh: line 37: doxygen: command not found
ninja: build stopped: subcommand failed.
Build of suite doc failed with Command '['ninja']' returned non-zero exit status 1.. See output above
```

(the build also reports exitcode 0 as in the case reported in #132)

### PR DESCRIPTION
**Pull request by yaqwsx**

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -198,8 +198,6 @@ run() {
     if [ -n "$ALTER_PROMPT" ]
     then
         PS1="🤖 $(configurationDesc $ROFI_BUILD_CONFIGURATION) $ORIGINAL_PS1"
-    else
-        PS1="$ORIGINAL_PS1"
```
[the remark this answers is not in the document]
Comment (yaqwsx) [reply setup.sh:202]: The exit code is propagated correctly. Just when you have a status of the last command shown in the prompt overwriting `PS1` will expand all variables and thus, you don't see when the command failed. This was the reason why #132 was reported.

**Review by vlstill — COMMENTED** [review]
The changes so fare seem good by me. However, `rmake doc` can still fail if doxigen extensions (such as `breathe`) are not installed:

```
Extension error:
Could not import extension breathe (exception: No module named 'breathe')
make[2]: *** [doc/CMakeFiles/doc.dir/build.make:98: web/index.html] Error 2
make[1]: *** [CMakeFiles/Makefile2:98: doc/CMakeFiles/doc.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
Build of suite doc failed with Command '['cmake', '--build', '/home/xstill/repo/rofi/build.Release/doc', '-j18']' returned non-zero exit status 2.. See output above
```

File: setup.sh [hunk 2]
```diff
@@ -92,8 +92,15 @@ setupIdf() {
...
             https://github.com/espressif/esp-idf.git $IDF_PATH
         $IDF_PATH/install.sh
+        IDF_POST_INSTALL=1
     fi
     source $IDF_PATH/export.sh
+
+    if [ $IDF_POST_INSTALL ]; then
+        # When we use IDF, it brings python venv; thus if we want to build doc,
+        # we have to install spinx and breathe into the venv
+        pip install sphinx breathe recommonmark sphinx_rtd_theme
```
[the remark this answers is not in the document]
Comment (vlstill) [reply setup.sh:108]: OK, in that case it seems reasonable to me to install the packages locally even if they are available in system.

**Review by vlstill — APPROVED** [review]

**Review by Ondraceq — CHANGES_REQUESTED** [review]

File: setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaqwsx) [reply setup.sh:202]: The general use is to open up a shell, set up the environment, work, kill the shell.

This is a workaround that doesn't properly work with the teardown or repeated invocation. In this way, it just does the least harm if you don't want to alter the shell. And I am not feeling like spending a lot of time on this here. True; we could probably backup PS1 only when it is changed. Would that seem sufficient to you?

[the remark this answers is not in the document]
Comment (Ondraceq) [reply setup.sh:202]: I'm more concerned about the fact that sometimes we show information that is not true, but I agree that it's not an important issue. Now I feel like creating an issue with the `won't fix` label documenting it is enough.

**Review by Ondraceq — APPROVED** [review]

File: setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (yaqwsx) [reply setup.sh:202]: ...or let's get rid of the shell alternation. Is there anyone actually using it?

**Comment by vlstill** [comment]
Just a note on shell alteration… it would be quite easy to export something like ROFI_PROMPT_ID and let the users who want to see it add '$ROFI_PROMPT_ID' to their PS1… I will cost almost nothing to have it there permanently and it would work safely for anyone who works with RoFI suite often and wants shell indication.

**Comment by yaqwsx** [comment]
I like the simplicity of that!

File: setup.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ondraceq) [reply setup.sh:202]: I'm using it, but don't know how much attention I actually put to it

**Pull request merged**